test-util/src/main/java/io/camunda/zeebe/test/util/asserts/EitherAssert.java

```diff
@@ -8,31 +8,121 @@
 package io.camunda.zeebe.test.util.asserts;
 
 import io.camunda.zeebe.util.Either;
+import org.assertj.core.api.AbstractAssert;
 import org.assertj.core.api.AbstractObjectAssert;
+import org.assertj.core.api.AssertFactory;
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.assertj.core.api.ObjectAssert;
+import org.assertj.core.api.ObjectAssertFactory;
 
+/**
+ * AssertJ assertions to help you test {@link Either} instances. Typically, you will check if it is
+ * of the expected side using {@link #isLeft()} or {@link #isRight()}, and then extract that value
+ * using {@link #left()}/{@link #left(AssertFactory)} and {@link #right()}/{@link
+ * #right(AssertFactory)} to assert properties on the actual result.
+ *
+ * @param <L> the left type
+ * @param <R> the right type
+ */
 public final class EitherAssert<L, R>
```

Comment:
:eyes: I just saw we have another `EitherAssert` in the util module's test classes...should we use that one? Should we use the one here? What's our idea?